File: code/169.求众数.go
/*
 * @lc app=leetcode.cn id=169 lang=golang
 *
 * [169] 求众数
 */
func majorityElement(nums []int) int {
	count := 0
	maj := nums[0]
	for _, n := range nums {
		if n == maj {
			count += 1
		} else {
			count -= 1
			if count == 0 {
				maj = n
				count = 1
			}
		}
	}
	return maj
}

File: code/169.求众数.rs
/*
 * @lc app=leetcode.cn id=169 lang=rust
 *
 * [169] 求众数
 */
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut count = 0;
        let mut maj = nums[0];
        for i in 0..nums.len() {
            if maj == nums[i] {
                count += 1;
            } else {
                count -= 1;
                if count == 0 {
                    maj = nums[i];
                    count = 1;
                }
            }
        }
        return maj;
    }
}

